src/Lachain.Storage/Repositories/LocalTransactionRepository.cs

```diff
@@ -31,23 +31,44 @@ public void SaveState(byte[] state)
 
         public void TryAddTransaction(TransactionReceipt receipt)
         {
-            if (_watchAddresses.Count(addr =>
-                    addr.Equals(receipt.Transaction.To) || addr.Equals(receipt.Transaction.From)) <= 0) return;
-
-            if (receipt.Transaction.Invocation.Length > 0)
+            try
             {
+                string[] signatures = {
+                    "transfer(address,uint256)", 
+                    "transferFrom(address,address,uint256)",
+                    "mint(address,uint256)",
+                    "burn(address,uint256)"
+                };
+                
                 var decoder = new ContractDecoderLtr(receipt.Transaction.Invocation.ToArray());
-                var res = decoder.Decode("uint160")[4] as UInt160;
-                if (res!.Equals(receipt.Transaction.To))
+                object[] decodedRes = {};
+                
+                foreach (var signature in signatures)
+                {
+                    decodedRes = decoder.Decode(signature);
+                    break;
+                }
+                
+                if (decodedRes.Length >= 5)
```

Comment:
Why we check 4-th place here only? As far as I can see from our signatures it should be decodedRes[0] or decodedRes[1] (we can check every member really  if it has address type)

Reply:
4th place because it is having the address

Reply:
this array hasn't 4 members for all of our signatures,  it has 2 members 3 of them and 3 members for  1 signature,  this code doesn't work as far as I can see. Search other usage of the Decode method in the code.